src/map_bindings.cpp

```diff
@@ -31,6 +31,199 @@ int py_call_strings(std::vector<std::string> assembly_list,
+        query_colours_map[query] = std::move(query_colours);
+    }
+
+    // get colour names
+    std::vector<std::string> input_colour_files = ccdbg.getColorNames();
+    std::vector<std::string> input_colour_pref;
+
+    // generate file prefixes for colours
+    for (const auto& file : input_colour_files)
+    {
+        std::filesystem::path p(file);
+        input_colour_pref.push_back(p.stem());
+    }
+
+    const std::pair<std::unordered_map<std::string, std::vector<bool>>, std::vector<std::string>> return_pair = std::make_pair(query_colours_map, input_colour_pref);
+
+    return return_pair;
+}
+
+std::pair<std::unordered_map<std::string, std::vector<bool>>,
+        std::vector<std::string>> py_uc_query_build (const std::string& infile1,
+                                                     const int& kmer,
+                                                     const std::string& query_file,
+                                                     size_t num_threads,
+                                                     bool is_ref,
+                                                     const bool write_graph,
+                                                     const std::string& infile2) {
+    // Set number of threads
+    if (num_threads < 1) {
+        num_threads = 1;
+    }
+
+    // read in compact coloured DBG
+    cout << "Building coloured compacted DBG..." << endl;
+
+    if (infile2 != "NA") {
+        is_ref = 0;
+    }
+
+    ColoredCDBG<> ccdbg;
+
+    // build graph, write graph if specified
+    size_t lastindex = infile1.find_last_of(".");
+    std::string outgraph = infile1.substr(0, lastindex);
+    ccdbg = buildGraph(infile1, infile2, is_ref, kmer, num_threads, false, write_graph, outgraph);
+
+    std::vector<std::string> query_list = parse_fasta(query_file);
+
+    // get the number of colours
+    const size_t nb_colours = ccdbg.getNbColors();
+
+    std::unordered_map<std::string, std::vector<bool>> query_colours_map;
+
+    cout << "Querying unitigs within population..." << endl;
+    for (const auto& query : query_list)
+    {
+        // run query of unitigs
+        std::vector<bool> query_colours = query_unitig(ccdbg, query, nb_colours);
+
+        // Add colours to map. If unitig not found, query colours will be empty
+        query_colours_map[query] = std::move(query_colours);
+    }
+
+    // get colour names
+    std::vector<std::string> input_colour_files = ccdbg.getColorNames();
+    std::vector<std::string> input_colour_pref;
+
+    // generate file prefixes for colours
+    for (const auto& file : input_colour_files)
+    {
+        std::filesystem::path p(file);
+        input_colour_pref.push_back(p.stem());
+    }
+
+    const std::pair<std::unordered_map<std::string, std::vector<bool>>, std::vector<std::string>> return_pair = std::make_pair(query_colours_map, input_colour_pref);
+
+    return return_pair;
+}
+
 PYBIND11_MODULE(map_strings, m)
```

Comment:
We should probably give the extension a more meaningful name. Perhaps `unitig_query`?

Reply:
Changed module name to unitig_query